pkg/network/tracer/connection/failure/matching.go

```diff
@@ -89,9 +101,8 @@ func (fc *FailedConns) upsertConn(failedConn *ebpf.FailedConn) {
 
 	stats, ok := fc.FailedConnMap[connTuple]
 	if !ok {
-		stats = &FailedConnStats{
-			CountByErrCode: make(map[uint32]uint32),
-		}
+		stats = fc.statsPool.Get()
+		stats.reset()
```

Comment:
If you are expecting reset to free up some memory, it would be better to reset before putting back into the pool.